js/src/components/shipping-rate-section/estimated-shipping-rates-card/rate-form-modals/edit-rate-form-modal.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import RateFormModal from './rate-form-modal.js';
+
+/**
+ * @typedef { import(".~/data/actions").CountryCode } CountryCode
+ * @typedef { import("../typedefs.js").ShippingRateGroup } ShippingRateGroup
+ */
+
+/**
+ * Form modal to edit or delete shipping rate group.
+ *
+ * @param {Object} props
+ * @param {Array<CountryCode>} props.countryOptions Array of country codes, to be used as options in AppCountrySelect.
+ * @param {ShippingRateGroup} props.initialValues Initial values for the form.
+ * @param {(values: ShippingRateGroup) => void} props.onSubmit Called when the shipping rate group is submitted.
+ * @param {() => void} props.onDelete Called when users clicked on the Delete button.
+ * @param {() => void} props.onRequestClose Callback to close the modal.
+ */
+const EditRateFormModal = ( {
+	countryOptions,
+	initialValues,
+	onSubmit,
+	onRequestClose = noop,
+	onDelete = noop,
```

Comment:
:nail_care: :grey_question: If we don't have a use case when we do not provide those, I'd not set the default value, to fail fast in case of an unexpected lack of those.